elastic_search/index.js

```diff
@@ -0,0 +1,135 @@
+/*
+This lambda function fetched search results from elastic search instance
+
+Request parameter has two nodes, state and secret.
+`state` node is empty for the first request. In this lambda function, cursor for entity `search` is stored in `state` object.
+`secrets` node contains the host name of elastic search instance 
+{
+    "state" : {},
+    "secrets" : {
+        "host" : "host_name"
+    }
+}
+
+Insert node contains records fetched using search query `bag`.
+Delete node contains records fetched using search query `puma`.
+
+Primary key for the records are [_id, _index, _type, sync_time]
+*/
+var RecordOperation = {
+    INSERT : 1,
+    DELETE : 2
+}
+
+exports.handler = (request, context, callback) => {
+    update(request.state, request.secrets, callback)
+}
+
+async function update(state, secrets, callback) {
+    let response = initializeResponse(state);
+     
+    let insertResponse = fetchSearchResult(state, secrets, 'products', 'product', 'Bag');
```

Comment:
is it case insensitive? we have `bag` and `puma` in comments above

Reply:
Yes, it is case insensitive. But I have changed to `Bag` and `Puma`